pubsub/flow_controller.go

```diff
@@ -176,3 +201,19 @@ func (f *flowController) bound(size int) int64 {
 func (f *flowController) count() int {
 	return int(atomic.LoadInt64(&f.countRemaining))
 }
+
+func (f *flowController) recordOutstandingMessages(ctx context.Context, n int64) {
+	if f.purpose == flowControllerPurposeTopic {
+		recordStat(ctx, PublisherOutstandingMessages, n)
+	} else {
+		recordStat(ctx, OutstandingMessages, n)
+	}
+}
+
+func (f *flowController) recordOutstandingBytes(ctx context.Context, n int64) {
+	if f.purpose == flowControllerPurposeTopic {
+		recordStat(ctx, PublisherOutstandingBytes, n)
+	} else {
```

Comment:
If we're not going to check the unknown value here, we should either remove that value above or explicitly check `f.purpose == flowControllerPurposeSubscription` here.